backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -888,8 +892,8 @@ private class InteropTransformer(val context: Context, override val irFile: IrFi
                     val target = targetSymbol.owner
                     val signatureTypes = target.allParameters.map { it.type } + target.returnType
 
-                    descriptor.typeParameters.forEachIndexed { index, typeParameterDescriptor ->
-                        val typeArgument = expression.getTypeArgument(typeParameterDescriptor)!!.toKotlinType()
+                    function.typeParameters.forEachIndexed { index, irTypeParameter ->
```

Comment:
Actually, `irTypeParameter.index = index`, I think the situation where `typeParameters` are returned shuffled is just ridiculous.

Reply:
Fixed